src/android/java/com/pspdfkit/cordova/action/annotation/ApplyInstantJsonAction.java

```diff
@@ -44,11 +43,13 @@ protected void execAction(JSONArray args, CallbackContext callbackContext) throw
 
     if (document != null) {
       final DataProvider dataProvider = new DocumentJsonDataProvider(annotationsJson);
-      DocumentJsonFormatter.importDocumentJsonAsync(document, dataProvider)
-          .subscribeOn(Schedulers.io())
-          .observeOn(AndroidSchedulers.mainThread())
-          .doOnError(e -> callbackContext.error(e.getMessage()))
-          .subscribe(() -> callbackContext.success());
+      cordovaPdfActivity.addSubscription(
```

Comment:
Based on this (and my previous comment about subscription handling inside the `CordovaPdfActivity`), I think we should extend the lifecycle of added subscriptions to the lifetime of the fragment (which is retained across configuration changes and handed from activity instance to instance). Only clearing subscription upon finishing the activity, should do the trick. Please also update the method's Javadoc, so it's explicitly said when the subscriptions are disposed.